### PR TITLE
Consolidate registration logic and add more tests

### PR DESCRIPTION
**Pull request by cbandy**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?
   - [x] Have you added automated tests?

**Type of Changes:**

 
 - [x] Testing enhancement
 - [x] Other

**Other Information**:

Issue: PGO-1105

### CODE REVIEW

File: internal/controller/postgrescluster/controller.go [hunk 1]
```diff
@@ -213,35 +210,11 @@ func (r *Reconciler) Reconcile(
...
-
-		if registrationRequiredStatusFound(cluster) {
-			cluster.Status.RegistrationRequired = nil
-			r.Recorder.Event(cluster, corev1.EventTypeNormal, "Token Verified", "Thank you for registering your installation of Crunchy Postgres for Kubernetes.")
-		}
+	if r.Registration != nil && r.Registration.Required(r.Recorder, cluster, &cluster.Status.Conditions) {
+		registration.SetAdvanceWarning(r.Recorder, cluster, &cluster.Status.Conditions)
 	}
+	cluster.Status.RegistrationRequired = nil
+	cluster.Status.TokenRequired = ""
```
Comment (tony-landreth) [inline internal/controller/postgrescluster/controller.go:217]: Will these lines eventually go away?

File: internal/registration/runner.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+		defer ticker.Stop()
+		ticks = ticker.C
+	}
+
+	log := logging.FromContext(ctx).WithValues("controller", "registration")
+
+	for {
+		select {
+		case <-ticks:
+			_, before := r.state()
```
Comment (tony-landreth) [inline internal/registration/runner.go:180]: :question: How does the Runner's state persist across a restart? Or does it have to?

File: internal/controller/postgrescluster/controller.go (hunk 1 above)
Comment (cbandy) [reply internal/controller/postgrescluster/controller.go:217]: Eventually. I noticed some odd behavior with `managedFields` when removing these from the CRD. We should zero them out (while in the CRD) before removing them entirely.

File: internal/registration/runner.go (hunk 2 above)
Comment (cbandy) [reply internal/registration/runner.go:180]: The state is loaded once during startup before the Manager calls this function:

https://github.com/CrunchyData/postgres-operator/blob/3315e0cb3bbb70b12bf635758c57b1b0ac1bdad3/cmd/postgres-operator/main.go#L105

**Review by tony-landreth — APPROVED** [review]

**Pull request merged**